src/cdk/tree/tree.ts

```diff
@@ -128,7 +128,10 @@ export class CdkTree<T> implements AfterContentChecked, CollectionViewer, OnDest
     new BehaviorSubject<{start: number, end: number}>({start: 0, end: Number.MAX_VALUE});
 
   constructor(private _differs: IterableDiffers,
-              private _changeDetectorRef: ChangeDetectorRef) {}
+              private _changeDetectorRef: ChangeDetectorRef,
+              private _elementRef: ElementRef<HTMLElement>) {
```

Comment:
Changing the constructor signature for `CdkTree` seems like it would be a breaking change since we do expect people will extend this class. Is there a way we can do this without changing the signature?

Reply:
The typical way we handle this is to make the new parameter optional and add a TODO

```ts
// @breaking-change 11.0.0 make this required
private _elementRef?: ElementRef<HTMLElement>
```

and then of course we need to handle the fact that the argument might be `undefined`. In this case I'm not sure how we could handle it being undefined though, since we've switched to applying the class this way...

Reply:
maybe instead we could do individual host bindings?

```ts
@HostBinding('class.cdk-tree') _cdkTreeClass = true;
```

Reply:
Probably, I can't think of another way we could do it

Reply:
**Solution:**
Add the class on the `host` property for `CdkTree` and have both `cdk-tree` and `mat-tree` classes on `MatTree`.

**Justification:**
- Cannot add the class as a `@HostBinding` as the class is not applied yet when the children nodes try to read the class off of it.
  - Moving the logic setting `aria-level` later causes further timing issues with `MatTree`
- Cannot directly add the class in the constructor as it is a breaking change (discussed above).

Added comment in the `MatTree` class for clarification